packages/workflow/src/features/events/handler.ts

```diff
@@ -83,7 +87,14 @@ function detectEvent(request: Hapi.Request): Events {
                 return Events.BIRTH_WAITING_VALIDATION
               }
             } else {
-              return Events.BIRTH_MARK_CERT
+              if (
+                hasRegisterScope(request) &&
+                hasCorrectionEncounterSection(firstEntry as fhir.Composition)
```

Comment:
The payload should have a registration number, as the records are to be corrected can be found in ready to print tab